src/Common/src/System/Security/Cryptography/RSAOpenSsl.cs

```diff
@@ -130,6 +130,54 @@ public override byte[] Decrypt(byte[] data, RSAEncryptionPadding padding)
             SafeRsaHandle key = _key.Value;
             CheckInvalidKey(key);
 
+            int keySizeBytes = Interop.Crypto.RsaSize(key);
+
+            // OpenSSL does not take a length value for the destination, so it can write out of bounds.
+            // To prevent the OOB write, decrypt into a temporary buffer.
+            if (destination.Length < keySizeBytes)
+            {
+                Span<byte> tmp = stackalloc byte[0];
+                byte[] rent = null;
+
+                // RSA up through 4096 stackalloc
+                if (keySizeBytes <= 512)
+                {
+                    tmp = stackalloc byte[keySizeBytes];
+                }
+                else
+                {
+                    rent = ArrayPool<byte>.Shared.Rent(keySizeBytes);
+                    tmp = rent;
+                }
+
+                bool ret = TryDecrypt(key, data, tmp, rsaPadding, oaepProcessor, out bytesWritten);
+
+                if (ret)
+                {
+                    Span<byte> tmpSlice = tmp.Slice(0, bytesWritten);
+
+                    if (bytesWritten > destination.Length)
+                    {
+                        ret = false;
+                        bytesWritten = 0;
+                    }
+                    else
+                    {
+                        tmpSlice.CopyTo(destination);
+                    }
+
+                    CryptographicOperations.ZeroMemory(tmpSlice);
+                }
+
+                if (rent != null)
+                {
+                    // Already cleared
```

Comment:
I assume this means that if TryDecrypt fails, it won't have written anything to the buffer?